ref(frames): Need indication in Issues UI that you can expand the frame

priscilawebdev opened the pull request:
closes https://app.asana.com/0/1182975495223914/1193781937756635


depends on https://github.com/getsentry/sentry/pull/21220

. Display some line actions, as "images loaded flag, switch to relative/absolute" when the user hovers the frame line
. Add a new line action to expand/collapse the functionName
. Update Action's icon color
. Update the icon of the button expand/collapse frame line


Now it looks like below:

![image](https://user-images.githubusercontent.com/29228205/95458215-282ea780-0972-11eb-95dc-ca20815598f3.png)

#sync-getsentry

priscilawebdev committed:
ref(frames): Add function expand indicator

markstory commented on src/sentry/static/sentry/app/components/events/interfaces/frame/line.tsx:
```diff
@@ -166,16 +179,19 @@ export class Line extends React.Component<Props, State> {
-      >
-        {this.state.isExpanded ? <IconSubtract size="8px" /> : <IconAdd size="8px" />}
-      </a>
+      <ToogleContextButtonWrapper>
```
```suggestion
      <ToggleContextButtonWrapper>
```

priscilawebdev replied:
Good catch!! Thanks!!

markstory commented on src/sentry/static/sentry/app/components/events/interfaces/frame/line.tsx:
```diff
@@ -166,16 +179,19 @@ export class Line extends React.Component<Props, State> {
+            direction={isExpanded ? 'up' : 'down'}
+            size="8px"
+          />
+        </ToogleContextButton>
```
```suggestion
        </ToggleContextButton>
```

markstory commented on src/sentry/static/sentry/app/components/events/interfaces/frame/line.tsx:
```diff
@@ -166,16 +179,19 @@ export class Line extends React.Component<Props, State> {
+            size="8px"
+          />
+        </ToogleContextButton>
+      </ToogleContextButtonWrapper>
```
```suggestion
      </ToggleContextButtonWrapper>
```

markstory commented on src/sentry/static/sentry/app/components/events/interfaces/frame/line.tsx:
```diff
@@ -471,4 +416,47 @@ const LeadHint = styled('div')`
+  transition: 0.1s all;
+`;
+
+const ToogleContextButtonWrapper = styled('span')`
```
```suggestion
const ToggleContextButtonWrapper = styled('span')`
```

markstory commented on src/sentry/static/sentry/app/components/events/interfaces/frame/line.tsx:
```diff
@@ -471,4 +416,47 @@ const LeadHint = styled('div')`
+`;
+
+// the Buton's label has the padding of 3px because the button size has to be 16x16 px.
+const ToogleContextButton = styled(Button)`
```
```suggestion
const ToggleContextButton = styled(Button)`
```

markstory commented on src/sentry/static/sentry/app/components/events/interfaces/frame/line.tsx:
```diff
@@ -471,4 +416,47 @@ const LeadHint = styled('div')`
+  margin-left: ${space(1)};
+`;
+
+// the Buton's label has the padding of 3px because the button size has to be 16x16 px.
```
```suggestion
// the Button's label has the padding of 3px because the button size has to be 16x16 px.
```

markstory commented on src/sentry/static/sentry/app/components/events/interfaces/frame/line.tsx:
```diff
@@ -471,4 +416,47 @@ const LeadHint = styled('div')`
+    flex-shrink: 0;
+  }
+  :hover {
+    ${PackageStatusIcon} {
+      visibility: visible;
+    }
+    ${AddressToggleIcon} {
+      visibility: visible;
+    }
+    ${FunctionNameToggleIcon} {
```
```suggestion
    ${PackageStatusIcon},
    ${AddressToggleIcon},
    ${FunctionNameToggleIcon} {
```

I think you can combine all these selectors to share a property list as they are the same.

matejminar commented:
Oops, I accidentally approved visual snapshot diffs.

Does the platform icon next to the stack trace still make sense now that mixed stack traces are deprecated? Can we please verify this with design?

priscilawebdev commented on src/sentry/static/sentry/app/components/events/interfaces/frame/line.tsx:
```diff
@@ -407,10 +367,11 @@ export class Line extends React.Component<Props, State> {
+      </StyledListItem>
     );
   }
 }
+export default withSentryAppComponents(Line, {componentType: 'stacktrace-link'});
```
```suggestion
}

export default withSentryAppComponents(Line, {componentType: 'stacktrace-link'});
```

jan-auer commented:
@priscilawebdev Could you post screenshots that compare before and after of a realistic event? You can use one of the example events in the integration test suite for this, or I can generate events for you.

priscilawebdev commented on src/sentry/static/sentry/app/components/events/interfaces/frame/symbol.tsx:
```diff
@@ -0,0 +1,152 @@
+      return t('Expand function details');
+    }
+
+    return t('Hide function details');
```
this should not happen...let me check that

jan-auer reviewed (commented):
Approving for staging deployment.

matejminar commented:
Are those visual changes in `acceptance/issue-details-cocoa.png`, `acceptance/issue-details-rust.png` there on purpose?

Should we make a visual snapshot of this new hovered frame so that we are sure there's no regression in the future?

priscilawebdev commented:
@matejminar  yes they are on purpose :) I fixed some alignment issues and added a new icon on the left side of the function name.. I'll create the visual snapshot that you suggested 😉

priscilawebdev commented on src/sentry/utils/pytest/selenium.py:
```diff
@@ -337,11 +337,12 @@ def snapshot(self, name, mobile_only=False):
                         )
 
-            with self.mobile_viewport():
+            if not desktop_only:
```
@billyvg Please is this change fine for you?

billyvg replied:
imo we should include the mobile snapshots, it’s not only for mobile devices -- it applies to desktop devices on smaller window sizes as well

priscilawebdev committed:
remove desktop condition

priscilawebdev committed:
added button class
